src/server/scripts/Northrend/IcecrownCitadel/boss_the_lich_king.cpp

```diff
@@ -1498,7 +1534,10 @@ class npc_valkyr_shadowguard : public CreatureScript
                 {
                     case POINT_DROP_PLAYER:
                         DoCastAOE(SPELL_EJECT_ALL_PASSENGERS);
-                        me->DespawnOrUnsummon(1000);
+                        if (IsHeroic())
+                            me->GetMotionMaster()->MoveTargetedHome();
```

Comment:
@ariel- There isnt one in the middle, other triggers are for finding to which location valkyr should move. What i would do is just define a const position (middle of the room) and then generate random near position to that position and move there, because they're not supposed to stack on top of each other